pandas/compat/numpy/function.py

```diff
@@ -245,3 +253,48 @@ def validate_transpose_for_generic(inst, kwargs):
             msg += " for {klass} instances".format(klass=klass)
 
         raise ValueError(msg)
+
+
+def validate_window_func(name, args, kwargs):
+    numpy_args = ('axis', 'dtype', 'out')
+    msg = ("numpy operations are not "
+           "valid with window objects. "
+           "Call the object's '{func}' "
+           "method directly instead".format(func=name))
```

Comment:
Hmm...not sure if that would be as straightforward because while there are methods like `.expanding(...)` or `.rolling(...)`, there is not `.window(...)` method.  This function that you highlighted above was supposed to be generic enough for ALL window objects.